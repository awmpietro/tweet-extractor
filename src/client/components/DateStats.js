import React from 'react';
import { Table, CardTitle } from 'reactstrap';
import PropTypes from 'prop-types';

const DateStats = ({stats}) => {

    const renderDates = () => {
        const dates = []
        for(let i in stats.dates){
          dates.push(
            <tr key={i}>
              <td>{i}</td>
              <td>{stats.dates[i]}</td>
            </tr>
          )
        }
        return dates;
    }

    return(
        <>
          <CardTitle><h4>Tweets by date</h4></CardTitle>
          <Table bordered striped>
            <thead>
              <tr>
                <th>Date</th>
                <th>Quantity</th>
              </tr>
            </thead>
            <tbody>
              {renderDates()}
            </tbody>
          </Table>
        </>
    )
}

DateStats.propTypes = {
    stats: PropTypes.object
};

export default DateStats;