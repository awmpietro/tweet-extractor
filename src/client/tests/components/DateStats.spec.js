import React from 'react';
import DateStats from '../../components/DateStats';
import renderer from 'react-test-renderer';

describe('DateStats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<DateStats stats={{}} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});