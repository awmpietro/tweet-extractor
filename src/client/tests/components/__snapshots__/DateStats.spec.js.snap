// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateStats snapshot renders 1`] = `
Array [
  <div
    className="card-title"
  >
    <h4>
      Tweets by date
    </h4>
  </div>,
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Quantity
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
